File: backend/utils/file_utils.py
import os
import glob2
from datetime import datetime

PROJECT_ROOT = os.path.abspath(os.path.join(os.path.dirname(__file__), '..', '..'))

def get_project_files(file_location):
    """Get all files (PDF, JPEG, PNG) for a project and return file information"""
    abs_path = os.path.abspath(file_location)
    file_types = [('pdf', 'pdf'), ('jpeg', 'img'), ('jpg', 'img'), ('png', 'img')]
    all_files = []
    most_recent = None

    for ext, ftype in file_types:
        pattern = os.path.join(abs_path, f"*.{ext}")
        files = glob2.glob(pattern)
        for f in files:
            ctime = os.path.getctime(f)
            file_info = {
                'path': f,
                'type': ftype,
                'ctime': ctime,
                'filename': os.path.basename(f),
                'rel_path': os.path.relpath(f, PROJECT_ROOT)
            }
            all_files.append(file_info)

            if (most_recent is None) or (ctime > most_recent['ctime']):
                most_recent = file_info

    # Sort files by creation time (newest first)
    all_files.sort(key=lambda x: x['ctime'], reverse=True)
    
    return {
        'all_files': all_files,
        'file_count': len(all_files),
        'most_recent': most_recent
    }

File: backend_node/utils/fileUtils.js
const fs = require('fs');
const path = require('path');
const glob = require('glob');

const PROJECT_ROOT = path.join(__dirname, '..', '..');

/**
 * Get all files (PDF, JPEG, PNG) for a project and return file information
 */
function getProjectFiles(fileLocation) {
    const absPath = path.resolve(fileLocation);
    const fileTypes = [
        { ext: 'pdf', type: 'pdf' },
        { ext: 'jpeg', type: 'img' },
        { ext: 'jpg', type: 'img' },
        { ext: 'png', type: 'img' }
    ];
    
    const allFiles = [];
    let mostRecent = null;

    for (const { ext, type } of fileTypes) {
        const pattern = path.join(absPath, `*.${ext}`);
        try {
            const files = glob.sync(pattern);
            
            for (const file of files) {
                const stats = fs.statSync(file);
                const ctime = stats.ctimeMs;
                
                const fileInfo = {
                    path: file,
                    type: type,
                    ctime: ctime,
                    filename: path.basename(file),
                    rel_path: path.relative(PROJECT_ROOT, file)
                };
                
                allFiles.push(fileInfo);

                if (!mostRecent || ctime > mostRecent.ctime) {
                    mostRecent = fileInfo;
                }
            }
        } catch (error) {
            // Continue if directory doesn't exist or other errors
            console.warn(`Warning: Could not scan directory ${absPath} for ${ext} files:`, error.message);
        }
    }

    // Sort files by creation time (newest first)
    allFiles.sort((a, b) => b.ctime - a.ctime);
    
    return {
        all_files: allFiles,
        file_count: allFiles.length,
        most_recent: mostRecent
    };
}

module.exports = {
    getProjectFiles,
    PROJECT_ROOT
};
